src/utils/global_config.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use std::cell::RefCell;
+
+/// Configuration values that affect most or all the
+/// components of the service.
+#[derive(Default, Debug, Copy, Clone)]
+pub struct GlobalConfig {
+    verbose_log: bool,
+}
+
+impl GlobalConfig {
+    /// Determine whether error logs should include detailed
+    /// information about the error
+    pub fn verbose_log() -> bool {
+        GLOBAL_CONFIG.0.borrow().verbose_log
+    }
+}
+
+struct GlobalConfigWrapper(RefCell<GlobalConfig>);
```

Comment:
We could also go full-safe and use a `RwLock`? Maybe the performence penalty of calling `read` is not too big?

Reply:
I'll drop the `GlobalConfigWrapper` and go with `AtomicBool` in the `GlobalConfig`, in the hope that we won't need non-integer-or-bool configs in the future.